lib/src/analysis_options.dart

```diff
@@ -25,74 +32,85 @@ class AnalysisOptions {
-        metricsExcludePatterns =
-            List.unmodifiable(metricsOptions[_metricsExcludeKey] as Iterable);
+      final excludeList = metricsOptions[_metricsExcludeKey];
+      if (excludeList != null &&
+          excludeList is Iterable<Object> &&
+          excludeList.every((element) => element is String)) {
+        metricsExcludePatterns = List<String>.unmodifiable(excludeList);
       }
 
-      final rulesNode = metricsOptions.nodes[_rulesKey];
-      if (rulesNode != null && rulesNode is YamlList) {
+      final rulesNode = metricsOptions[_rulesKey];
+      if (rulesNode != null && rulesNode is Iterable<Object>) {
         rules = Map.unmodifiable(Map<String, Map<String, Object>>.fromEntries([
-          ...rulesNode.nodes
-              .whereType<YamlScalar>()
-              .map((node) => MapEntry(node.value as String, {})),
-          ...rulesNode.nodes
-              .whereType<YamlMap>()
+          ...rulesNode.whereType<String>().map((node) => MapEntry(node, {})),
+          ...rulesNode
+              .whereType<Map<String, Object>>()
               .where((node) =>
-                  node.keys.length == 1 && node.values.first is YamlMap)
-              .map((node) => MapEntry(node.keys.cast<String>().first,
-                  (node.values.first as Map).cast<String, Object>())),
+                  node.keys.length == 1 &&
+                  node.values.first is Map<String, Object>)
+              .map((node) => MapEntry(
+                  node.keys.first, node.values.first as Map<String, Object>)),
         ]));
-      } else if (rulesNode != null && rulesNode is YamlMap) {
+      } else if (rulesNode != null && rulesNode is Map<String, Object>) {
         rules = Map.unmodifiable(Map<String, Map<String, Object>>.fromEntries([
-          ...rulesNode.keys.cast<String>().where((key) {
-            final node = rulesNode.nodes[key];
+          ...rulesNode.keys.where((key) {
+            final scalar = rulesNode[key];
 
-            return (node is YamlScalar && node.value is bool) &&
-                (node.value as bool);
+            return scalar != null && scalar is bool && scalar;
           }).map((key) => MapEntry(key, {})),
-          ...rulesNode.keys
-              .cast<String>()
-              .where((key) => rulesNode.nodes[key] is YamlMap)
-              .map((key) => MapEntry(key,
-                  (rulesNode.nodes[key].value as Map).cast<String, Object>())),
+          ...rulesNode.keys.where((key) {
+            final node = rulesNode[key];
+
+            return node != null && node is Map<String, Object>;
+          }).map((key) => MapEntry(key, rulesNode[key] as Map<String, Object>)),
         ]));
       }
     }
 
+    final analyzerOptions = configMap[_analyzerKey];
+    if (analyzerOptions != null && analyzerOptions is Map<String, Object>) {
+      final excludeList = analyzerOptions[_excludeKey];
+      if (excludeList != null && excludeList is List<Object>) {
```

Comment:
Can it be `excludeList is Iterable<Object>` instead of a `List`?

Reply:
fixed